planner/core/rule_partition_processor.go

```diff
@@ -472,6 +478,33 @@ func partitionRangeForOrExpr(sctx sessionctx.Context, expr1, expr2 expression.Ex
 	return tmp1.union(tmp2)
 }
 
+func partitionRangeForInExpr(sctx sessionctx.Context, args []expression.Expression,
+	pruner *rangePruner) partitionRangeOR {
+	if len(args) <= 1 {
```

Comment:
Add a test case for this check.

Reply:
The code is impossible to run here, because 

```
col in (xxx) 
```

represented as expression is `In(col, xxx)`

`len(args) <= 1` means  `col in ()`

It's impossible to construct such a query:

```
mysql> select * from t where a in ();
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 29 near ")" 
```

Reply:
So we do not need this check actually?

Reply:
I'll remove it